src/neptune/metadata_containers/safe_container.py

```diff
@@ -13,34 +13,33 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-__all__ = ["SafeContainer"]
+__all__ = ["decorator_safety"]
 
+import functools
 from typing import (
     Any,
     Callable,
 )
 
 from neptune.internal.utils.logger import logger
-from neptune.metadata_containers import MetadataContainer
+import inspect
 
 
-class SafeMethodWrapper:
-    def __init__(self, method: Callable):
-        self._method = method
-
-    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+def _safe_function(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
         try:
-            return self._method(*args, **kwargs)
+            return func(*args, **kwargs)
         except Exception:
-            logger.exception(f"Exception in method {self._method}")
-
-    def __getattr__(self, item: str) -> Any:
-        return getattr(self._api_method, item)
+            logger.exception(f"Exception in method ")
 
+    return wrapper
 
-class SafeContainer:
-    def __init__(self, container: MetadataContainer):
-        self._container = container
 
-    def __getattr__(self, item: str) -> Any:
-        return SafeMethodWrapper(getattr(self._container, item))
+def decorator_safety(cls):
+    for name, method in inspect.getmembers(cls):
+        if (not inspect.ismethod(method) and not inspect.isfunction(method)) or inspect.isbuiltin(method):
```

Comment:
`or inspect.isbuiltin(method)`

So will it decorate `__setitem__` etc. or not?

Reply:
List of decorated methods: 
```
Decorating function __class_getitem__
Decorating function __delitem__
Decorating function __enter__
Decorating function __exit__
Decorating function __getattr__
Decorating function __getitem__
Decorating function __init__
Decorating function __setitem__
Decorating function __subclasshook__
Decorating function _before_fork
Decorating function _define_attribute
Decorating function _fetch_entries
Decorating function _get_callback
Decorating function _get_or_create_api_object
Decorating function _get_root_handler
Decorating function _get_subpath_suggestions
Decorating function _handle_fork_in_child
Decorating function _handle_fork_in_parent
Decorating function _ipython_key_completions_
Decorating function _pop_impl
Decorating function _prepare_background_jobs
Decorating function _prepare_background_jobs_if_non_read_only
Decorating function _print_structure_impl
Decorating function _raise_if_stopped
Decorating function _shutdown_hook
Decorating function _startup
Decorating function _write_initial_attributes
Decorating function assign
Decorating function define
Decorating function exists
Decorating function fetch
Decorating function get_attribute
Decorating function get_root_object
Decorating function get_state
Decorating function get_structure
Decorating function get_url
Decorating function lock
Decorating function ping
Decorating function pop
Decorating function print_structure
Decorating function set_attribute
Decorating function start
Decorating function stop
Decorating function sync
Decorating function wait
```